### PR TITLE
Setting Logger with configurable debug option

### PR DESCRIPTION
**Pull request by varun1524**
Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/runtime/config.go [hunk 1]
```diff
@@ -40,4 +41,10 @@ func (c *Config) BindFlags() {
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.",
 	)
+	flag.BoolVar(
+		&c.EnableDevModeLogging, "enable-dev-mode-logging",
+		false,
+		"Configures the logger to use a Zap development config (stacktraces on warnings, no sampling), "+
```
Comment (nithu0115) [inline pkg/runtime/config.go:47]: Nit: Probably, we can add more info about logLevel as well, something like this: `Development Mode, defaults(encoder=consoleEncoder,logLevel=Debug,stackTraceLevel=Warn, no sampling) + "Production Mode defaults(encoder=jsonEncoder,logLevel=Info,stackTraceLevel=Error), sampling")`

Comment (varun1524) [reply pkg/runtime/config.go:47]: sure. looks better

File: services/bookstore/cmd/controller/main.go [hunk 2]
```diff
@@ -42,6 +43,10 @@ func main() {
 	var ackCfg ackrt.Config
 	ackCfg.BindFlags()
 	flag.Parse()
+	zapOptions := zap.Options{
+		Development: ackCfg.EnableDevModeLogging,
+	}
+	ctrlrt.SetLogger(zap.New(zap.UseFlagOptions(&zapOptions)))
```
Comment (jaypipes) [inline services/bookstore/cmd/controller/main.go:49]: We will want to hide all of this inside an `ackCfg.SetupLogger()` method. The benefits of doing that are:

* No need to import the zap library above
* There will only be a single call added to the main.go template and all generated controllers (easier maintenance)

File: pkg/runtime/config.go [hunk 3]
```diff
@@ -40,4 +41,10 @@ func (c *Config) BindFlags() {
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.",
 	)
+	flag.BoolVar(
+		&c.EnableDevModeLogging, "enable-dev-mode-logging",
```
Comment (jaypipes) [inline pkg/runtime/config.go:45]: I'd prefer naming the option `enable-development-logging`. The shortened term "dev" is easily confused with "device" :)

Comment (varun1524) [reply pkg/runtime/config.go]: done,

File: services/petstore/cmd/controller/main.go [hunk 4]
```diff
@@ -14,8 +14,6 @@
 package main
 
 import (
-	"os"
```
Comment (jaypipes) [inline services/petstore/cmd/controller/main.go:17]: Looks like this was accidentally removed, though?

Comment (varun1524) [reply services/petstore/cmd/controller/main.go]: yea fixed

**Pull request merged**